### PR TITLE
Mvc autofac

### PR DESCRIPTION
**Pull request by AntsiferauGodel**

### CODE REVIEW

File: Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        [Authorize]
+        public ActionResult AdvertisementsForCurrentUser(string userId)
+        {
+            IList<AdvertisementHeaderView> headers = new List<AdvertisementHeaderView>();
+            var advertisements = advertisementService.GetAdvertisementsForCurrentUser(userId);
+            for (int i = 0; i < advertisements.Count; i++)
+            {
+                headers.Add(AdvertisementEntityToViewConverter.ConvertToAdvertisementHeader(advertisements[i]));
+            }
```
Comment (HannaKeryna) [inline Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs]: What do you think about:
var headers = advertisements.Select(AdvertisementEntityToViewConverter.ConvertToAdvertisementHeader).ToList();

Please analyze all methods

File: Lab08.MVC/Lab08.MVC.Data/Repositories/AdvertisementManager.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Lab08.MVC.Data.DbContext;
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Data/Repositories/AdvertisementManager.cs]: I can't they are all in use....)

File: Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs (hunk 1 above)
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Web/Controllers/HomeController.cs]: cool! didnt knew about that thing..)

File: Lab08.MVC/Lab08.MVC.Business/Services/AdvertisementService.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        public IUnitOfWork UnitOfWork { get; set; }
+
+        public AdvertisementService(IUnitOfWork unitOfWork)
+        {
+            UnitOfWork = unitOfWork;
+        }
+
+        public void Create(AdvertisementView advertisement)
+        {
+            UnitOfWork.AdvertisementManager.Create(AdvertisementViewToEntityConverter.Convert(advertisement)); // TODO: INVESTIGATE
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Lab08.MVC/Lab08.MVC.Business/Services/AdvertisementService.cs]: done, thx

**Pull request merged**